printing/ppconstr.ml

```diff
@@ -725,7 +725,7 @@ let tag_var = tag Tag.variable
 
   let default_term_pr = {
     pr_constr_expr   = pr_simpleconstr;
-    pr_lconstr_expr  = (fun env sigma -> pr_expr env sigma ltop);
+    pr_lconstr_expr  = (fun env sigma -> pr_expr env sigma lsimpleconstr);
```

Comment:
I am fairly sure this is wrong, since it's not the same level at all and it will wreak havoc in printing. More generally, this file is a central one for Coq term printing, if there is a bug in SSR printing with very high probability the fix should be circumscribed there.

Reply:
It may well be wrong, but this is not a fix for a bug in ssr. It is a fix for a broad range of bugs, of which tihs one is a prototype:
```
Ltac t1 x := replace (x x) with (x x).
```
In general, all tactics that use `wit_constr` have a bug in printing such that there are no parentheses around the outermost application of the term.

Reply:
Then the problem is that those tactics are using the wrong generic argument.

Reply:
Ah, so you are saying that for all tactics that have this problem, I should change `constr(...)` into `uconstr(..)`? I assumed that there is a reason why they are different. For example, `replace` uses `uconstr` for its first argument and `constr` for its second. I assume there is a reason fot that?

Reply:
I should also note that I tested this on all tactics of the standar library, and all of them are parsable after this change. (I did not check if they parse to the corret syntax tree though)

Reply:
But now that I'm looking at it, this is indeed printing wrong now:
```
Parameter t : bool -> Prop.
Goal t true. (* Prints (t true) *)
```

Reply:
Not `uconstr`, which is for untyped terms. I believe that it's probably `lconstr` or something similar.

Reply:
@ppedrot is right. The bug is elsewhere, presumably in `pptactic.ml` where `register_print0` should bind `wit_constr` to `pr_constr_expr` and `pr_glob_constr_pptac` (rather than `pr_lconstr_expr` and `pr_lglob_constr_pptac`). Presumably a bad copy-paste of mine in be133d7bc9.

Reply:
Okay, I made a second attempt.